Plugin.pm

```diff
@@ -1800,4 +1837,87 @@ sub _localDate {  # convert input date string in format YYYY-MM-DD to localized
 	return strftime(preferences('server')->get('shortdateFormat'), 0, 0, 0, $dt[2], $dt[1] - 1, $dt[0] - 1900);
 }
 
+sub addRecentSearch {
+	my $search = shift;
+	main::DEBUGLOG && $log->is_debug && $log->debug("++addRecentSearch");
+
+	my $list = $prefs->get('qobuz_recent_search') || [];
+
+	$list = [ grep { $_ ne $search } @$list ];
+
+	push @$list, $search;
+
+	$list = [ @$list[(-1 * MAX_RECENT)..-1] ] if scalar @$list > MAX_RECENT;
```

Comment:
You could rewrite this as `$list = [ splice(@$list, MAX_RECENT) ] if ...`

Reply:
This didn't work - the array is populated oldest to newest (and then presented in reverse order) and we want to remove the oldest elements over the limit. a `shift` did the job.

Reply:
Yes, I have that page in my bookmarks - it's where I found the solution: `shift(@a) == splice(@a,0,1)`, given that in use we're never going to have more than MAX_RECENT + 1 elements in the array. I only replaced the `if` with `while` to cater for the extremely unlikely case where MAX_RECENT is made smaller in a future release of the plug-in.